flink-autoscaler/src/main/java/org/apache/flink/autoscaler/tuning/MemoryTuning.java

```diff
@@ -243,28 +244,36 @@ private static MemorySize adjustNetworkMemory(
             Configuration config,
             MemoryBudget memBudget) {
 
-        final long buffersPerChannel =
+        final int buffersPerChannel =
                 config.get(NettyShuffleEnvironmentOptions.NETWORK_BUFFERS_PER_CHANNEL);
-        final long floatingBuffers =
+        final int floatingBuffers =
                 config.get(NettyShuffleEnvironmentOptions.NETWORK_EXTRA_BUFFERS_PER_GATE);
         final long memorySegmentBytes =
                 config.get(TaskManagerOptions.MEMORY_SEGMENT_SIZE).getBytes();
 
         long maxNetworkMemory = 0;
         for (VertexInfo vertexInfo : jobTopology.getVertexInfos().values()) {
             // Add max amount of memory for each input gate
-            for (JobVertexID input : vertexInfo.getInputs()) {
-                int inputParallelism = updatedParallelisms.get(input);
+            for (Map.Entry<JobVertexID, String> inputEntry : vertexInfo.getInputs().entrySet()) {
                 maxNetworkMemory +=
-                        (inputParallelism * buffersPerChannel + floatingBuffers)
+                        calculateNetworkSegmentNumber(
+                                        updatedParallelisms.get(vertexInfo.getId()),
+                                        updatedParallelisms.get(inputEntry.getKey()),
+                                        inputEntry.getValue(),
```

Comment:
For readability, can we add a variable?

```suggestion
                                        shipStrategy,
```